### ISSUE TITLE
Frequent test failure in pipeline-pgroup

### ISSUE THREAD

**Issue by zanchey** [fish-shell/fish-shell#6693]
Travis and the OBS builds are regularly, though not always, failing the pipeline-pgroup check with:

```
Failure in checks/pipeline-pgroup.fish:

  The CHECK on line 51 wants:
    {{(\d+) (?!\1)\d+}}

  which failed to match line stdout:3:
    15362

  additional output:
    15360 

  when running command:
    env fth=../test/root/bin/fish_test_helper ../test/root/bin/fish checks/pipeline-pgroup.fish
```

This block in particular fails:

```fish
# Ensure that if a background job launches another background job, that they have different pgroups. 
# The pipeline here will arrange for the two pgroups to be printed on the same line, like:           
#   123 124
# Our regex will capture the first pgroup and use a negative lookahead on the second.                
status job-control full                                                                              
$fth print_pgrp | begin                                                                                                               tr \n ' '                                                                                        
    $fth print_pgrp &                                                                                
end &                                                                                                
wait                                                                                                 
# CHECK: {{(\d+) (?!\1)\d+}}
```

I can reproduce this locally, though it takes many iterations:

```fish
cd build/tests
set s 0
while ../littlecheck.py -s fish=../test/root/bin/fish -s fish_test_helper=../test/root/bin/fish_test_helper checks/pipeline-pgroup.fish
    set s (math $s + 1)
end
echo $s
```

(usually < 100 iterations)

**Comment by zanchey** [comment]
I wonder if this is the same as the reordering we see in #6573?

**Comment by zanchey** [comment]
Hilariously*, the problem disappears under if the fish process is being `strace`d.

**Comment by zanchey** [comment]
This now happens on 75+% of builds on OBS and in the PPA, I think since some changes around pgroups were merged recently.

**Comment by ridiculousfish** [comment]
It looks like when it fails, the pgroups are being printed on different lines - somehow a newline is sneaking in.

**Comment by faho** [comment]
Note: e390ee1 disables this test because it completely obscures everything else.

**Comment by faho** [comment]
Fixed by a698b1f96ca162c257df4a573f3d7817e5eb8f44.
